### PR TITLE
Fix Rack integration overriding 'notice[:params]'

### LINKED ISSUE #715: [BUG] HttpParamsFilter overrides custom params.
When putting extra params in an exception using `to_airbrake` or when using the `add_filter` like so:
```ruby
Airbrake.add_filter do |notice|
  notice[:params][:xxxx] = "1234"
end
```
They get overriden if the Airbrake notice contains the rack request in filtering (controller errors, etc) here: https://github.com/airbrake/airbrake-ruby/blob/master/lib/airbrake-ruby/notifier.rb#L124
Because the custom filter can be applied before `HttpParamsFilter`

Therefore this line here: https://github.com/airbrake/airbrake/blob/master/lib/airbrake/rack/http_params_filter.rb#L13
```ruby
notice[:params] = request.params
```
needs to be:
```ruby
notice[:params].merge! request.params
```

### PR DESCRIPTION
**Pull request by kyrylo**
Fixes #715 (HttpParamsFilter overrides custom params)

### CODE REVIEW

File: lib/airbrake/rack/http_params_filter.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class HttpParamsFilter
       def call(notice)
         return unless (request = notice.stash[:rack_request])
 
-        notice[:params] = request.params
+        notice[:params].merge!(request.params)
```
Comment (PeterWuMC) [inline lib/airbrake/rack/http_params_filter.rb]: I would imagine you want it to be reverse merge.
so your filter can mask the request params from sensitive information

**Comment by vmihailenco** [comment]
Why not to save them under 2 separate keys?

File: lib/airbrake/rack/http_params_filter.rb (hunk 1 above)
Comment (dragosmiron) [inline lib/airbrake/rack/http_params_filter.rb]: i think there is a separate filter that does the masking?

Comment (kyrylo) [inline lib/airbrake/rack/http_params_filter.rb]: Yeah, we have a filter that does this dirty job, so we should be good here.

**Comment by kyrylo** [comment]
> @vmihailenco 
> Why not to save them under 2 separate keys?

`notice[:params][:httpParams]`?

**Comment by kyrylo** [comment]
Merging as is for now.

**Pull request merged**